tracker_linux.go

```diff
@@ -140,12 +140,12 @@ func (t *linuxProcessTracker) doCleanupByEnvironmentVariable() error {
 // cleanupProcess terminates the process given by its PID. If the process has
 // already terminated, this will not return an error.
 func cleanupProcess(pid int) error {
-	catcher := grip.NewBasicCatcher()
 	// A process returns syscall.ESRCH if it already terminated.
-	if err := syscall.Kill(pid, syscall.SIGTERM); err != syscall.ESRCH {
-		catcher.Add(errors.Wrapf(err, "failed to send sigterm to process with pid '%d'", pid))
-		catcher.Add(errors.Wrapf(syscall.Kill(pid, syscall.SIGKILL), "failed to send sigkill to process with pid '%d'", pid))
-	}
+	catcher := grip.NewBasicCatcher()
+	err := syscall.Kill(pid, syscall.SIGTERM)
+	catcher.AddWhen(err != syscall.ESRCH, errors.Wrapf(err, "sending sigterm to process with PID '%d'", pid))
+	err = syscall.Kill(pid, syscall.SIGKILL)
```

Comment:
Shouldn't we check that there was an error before calling this?

Reply:
We could, but we're mostly interested in killing the process. Since the process can ignore SIGTERM, we send SIGKILL in case it doesn't really want to shut down.